roaring/roaring.go

```diff
@@ -2138,15 +2138,25 @@ func intersectArrayBitmap(a, b *Container) *Container {
 }
 
 func intersectBitmapBitmap(a, b *Container) *Container {
-	output := &Container{bitmap: make([]uint64, bitmapN), containerType: containerBitmap}
-
-	for i := range a.bitmap {
-		v := a.bitmap[i] & b.bitmap[i]
-		output.bitmap[i] = v
-		output.n += int(popcount(v))
+	// local variables added to prevent BCE checks in loop
+	// see https://go101.org/article/bounds-check-elimination.html
+	var (
+		ab  = a.bitmap[:bitmapN]
+		bb  = b.bitmap[:bitmapN]
+		buf = make([]uint64, bitmapN)
+		ob  = buf[:bitmapN]
+		n   int
+	)
+	for i := 0; i < bitmapN; i++ {
+		ob[i] = ab[i] & bb[i]
+		n += int(popcount(ob[i]))
+	}
 
+	output := &Container{
+		bitmap:        ob,
+		n:             n,
+		containerType: containerBitmap,
 	}
-	output.optimize()
```

Comment:
removing this optimize seems like a pretty big change that could affect performance differently for different workloads. Did you have any thoughts on this?

Reply:
My rationale was the overhead of the optimize call was making benchmark comparison of bounds checks inconclusive.  The optimize call was overshadowing the bce by a margin large enough to make bce almost a rounding error